notifications_utils/field.py

```diff
@@ -166,10 +166,23 @@ def replace_match(self, match):
             return placeholder.get_conditional_body(replacement)
 
         if placeholder.is_unsafe():
-            return "SANITISED"
+            return self.sanitise_replacement_unsafe(replacement)
 
         return replacement
 
+    def sanitise_replacement_unsafe(self, replacement: str):
+        # if the replacement contains a link consider it all compromised
+        html_pattern = re.compile(
+            r'https?://\S+|www\.\S+'
+        )
```

Comment:
Consider reusing `formatters.url`:

https://github.com/alphagov/notifications-utils/blob/bf94e62d250b12fa0e5dfe50e50ff06ae5f831eb/notifications_utils/formatters.py#L47-L55

Reply:
ah cool I didn't realise we have that!